python/cac_tripplanner/cac_tripplanner/views.py

```diff
@@ -1,9 +1,74 @@
+import json
+
+from django.views.generic import View
+from django.http import HttpRequest, HttpResponse
 from django.shortcuts import render_to_response
 from django.template import RequestContext
 
+from django.contrib.gis.geos import GEOSGeometry, MultiPolygon
+from requests import get
+
+from destinations.models import Destination
+
 def home(request):
     print request.META['REMOTE_ADDR']
     return render_to_response('home.html', context_instance=RequestContext(request))
 
+
 def map(request):
     return render_to_response('map.html', context_instance=RequestContext(request))
+
+
+class Reachable(View):
+    """Class based view for finding destinations of interest within the calculated isochrone"""
+    # TODO: make decisions on acceptable ranges of values that this endpoint will support
+
+    otp_router = 'default'
+    otp_url = 'http://192.168.8.26:8080/otp/routers/{router}/isochrone'.format(router=otp_router)
+    algorithm = 'accSampling'
+
+    def isochrone(self, lat, lng, mode, date, time, max_travel_time, max_walk_distance):
+        """Make request to Open Trip Planner for isochrone geometry - convert it to GDAL Polygon"""
+        payload = {
+            'routerId': self.otp_router,
+            'fromPlace': lat + ',' + lng,
+            'mode': (',').join(mode),
+            'date': date,
+            'time': time,
+            'cutoffSec': max_travel_time,
+            'maxWalkDistance': max_walk_distance,
+            'algorithm': self.algorithm
+        }
+        isochrone_response = get(self.otp_url, params=payload)
```

Comment:
It would be nice not to have this call to an outside service in the view although this is definitely simpler for the moment since I think the alternative is modifying OTP.